### ISSUE TITLE
[Bug Report] SslProtocolType is not valid for net472

### ISSUE THREAD

**Issue by eddynaka** [Azure/azure-iot-sdk-csharp#2056]
# Context

- **OS, version, SKU and CPU architecture used:** Windows 10
- **Application's .NET Target Framework :** net472
- **Device:** Cosmos
- **SDK version used:** 1.37.2

## Description of the issue
It fails for specific connection strings. Couldn't figure out any pattern.

## Code sample exhibiting the issue
```csharp
using var client = DeviceClient.CreateFromConnectionString(connection);
                options.TryGetNonEmptyValue("retry", out string retry);
                client.OperationTimeoutInMilliseconds = (retry == bool.TrueString) ? 15000U : 3000U;
                client.OpenAsync().GetAwaiter().GetResult();
```

## Console log of the issue
Exception message: The specified value is not valid in the 'SslProtocolType' enumeration.\r\nParameter name: sslProtocolType

**Comment by timtay-microsoft** [comment]
Can you provide the full stacktrace for this exception?

**Comment by timtay-microsoft** [comment]
I talked with @eddynaka offline about this, and switching to MQTT unblocked him so I'm closing this issue.

**Comment by timtay-microsoft** [comment]
While the user is unblocked here, this still appears to be a bug in the SDK since AMQP should work for net472 applications. Re-opening this issue because of that

**Comment by vinagesh** [comment]
@eddynaka - I tried to repro this but was not able to. Can you help me with the steps to repro this?

**Comment by eddynaka** [comment]
@vinagesh, I'm running this in cosmos, which uses netframework 4.7.2.
Besides that, it's hard to debug, since it's just like a function/batch...

**Comment by jamdavi** [comment]
Spoke with @eddynaka offline and found that they are loading their assemblies into an environment they do not have control over via a reflection mechanism they did not write.

The issue is we're unsure of what target framework is installed on these machines and what (if any) settings might be affecting the SslProtocols selection.

For now Eddy is using a workaround and setting the transport to Http1. One open item for Eddy is to place the following code snippet to force Tls12 via our TlsVersions class above his client initialization.

```c#
TlsVersions.Instance.SetMinimumTlsVersions(SslProtocols.Tls12);
```

**Comment by vinagesh** [comment]
Customer is unblocked and not been able to repro it. If they find issues, they will report and create a new issue.
